### PR TITLE
feature: TCC adapter for Dubbo And Sofa reference annotation

### PR DESCRIPTION
**Pull request by q294881866**
### Ⅰ. Describe what this PR did
Support Dubbo @Reference and sofa @SofaReference  annotation for TCC mode

### CODE REVIEW

File: spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+     * @param beanName
+     * @param annotation
+     */
+    protected void process(Object bean, String beanName, Class<? extends Annotation> annotation) {
+        if (Objects.isNull(annotation) || PROXIED_SET.contains(beanName)) {
+            return;
+        }
+
+        ReflectionUtils.doWithFields(bean.getClass(), field -> {
+            Annotation reference = field.getAnnotation(annotation);
```
Comment (slievrly) [inline spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:75]: if necessary? FieldFilter check the filed contains annotation.

File: spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            if (reference == null) {
+                return;
+            }
+
+            addTccAdvise(bean, beanName, field, field.getType());
```
Comment (slievrly) [inline spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:80]: why not getInterface()?

Comment (q294881866) [reply spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:80]: Previously, it was implemented with an interface, but there were many repetitions. In fact, it's the only one that's different. There's no need to use an interface
` static {
        ANNOTATIONS.add(loadAnnotation("org.apache.dubbo.config.annotation.Reference"));
        ANNOTATIONS.add(loadAnnotation("com.alipay.sofa.runtime.api.annotation.SofaReference"));
        ANNOTATIONS.add(loadAnnotation( ...
    }`

File: spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java (hunk 1 above)
Comment (q294881866) [reply spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:75]: The code impossible to reuse.

Comment (q294881866) [reply spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:75]: > if necessary? FieldFilter check the filed contains annotation.
The code impossible to reuse.

Comment (slievrly) [reply spring/src/main/java/io/seata/spring/tcc/TccAnnotationProcessor.java:75]: What I mean is that reference can never null， `field.isAnnotationPresent(annotation)`

**Pull request merged**